dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -510,6 +543,10 @@ private ContextVariables GetNextStepVariables(ContextVariables variables, Plan s
             {
                 stepVariables.Set(item.Key, value);
             }
+            else if (!string.IsNullOrEmpty(expandedValue) && !stepVariables.Get(item.Key, out var _))
```

Comment:
the check on 528 should take care of this part
  !stepVariables.Get(item.Key, out var _)

Reply:
I was worried about `string.Empty` values, but I'm letting go of that for now. If someone sets something to an empty string, we'll treat it as valid.